### ISSUE TITLE
Xamarin.Mac + log4net - Exception occurred resolving a log provider

### ISSUE THREAD

**Issue by tofutim** [damianh/LibLog#128]
I am able to add log4net to a Xamarin.Mac (2.0/xamarin mac mobile framework) as well as LibLog. However, while log4net works (though strangely requires creation of a repository unlike windows), I get this from LibLog
```
Exception occurred resolving a log provider. Logging for this assembly testLog2, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null is disabled. System.ArgumentNullException: Value cannot be null.
Parameter name: method
  at System.Dynamic.Utils.ContractUtils.RequiresNotNull (System.Object value, System.String paramName) [0x00006] in /Library/Frameworks/Xamarin.Mac.framework/Versions/3.0.0.367/src/mono/mcs/class/dlr/Runtime/Microsoft.Scripting.Core/Utils/ContractUtils.cs:58 
  at System.Linq.Expressions.Expression.Call (System.Linq.Expressions.Expression instance, System.Reflection.MethodInfo method, System.Collections.Generic.IEnumerable`1[T] arguments) [0x00000] in /Library/Frameworks/Xamarin.Mac.framework/Versions/3.0.0.367/src/mono/mcs/class/dlr/Runtime/Microsoft.Scripting.Core/Ast/MethodCallExpression.cs:787 
  at System.Linq.Expressions.Expression.Call (System.Linq.Expressions.Expression instance, System.Reflection.MethodInfo method, System.Linq.Expressions.Expression[] arguments) [0x00000] in /Library/Frameworks/Xamarin.Mac.framework/Versions/3.0.0.367/src/mono/mcs/class/dlr/Runtime/Microsoft.Scripting.Core/Ast/MethodCallExpression.cs:674 
  at testLog2.Logging.LogProviders.Log4NetLogProvider.GetGetLoggerMethodCall () [0x00047] in /Users/tim/Projects/testLog/testLog2/App_Packages/LibLog.4.2/LibLog.cs:1224 
  at testLog2.Logging.LogProviders.Log4NetLogProvider..ctor () [0x0001e] in /Users/tim/Projects/testLog/testLog2/App_Packages/LibLog.4.2/LibLog.cs:1132 
  at testLog2.Logging.LogProvider.<LogProviderResolvers>m__2 () [0x00000] in /Users/tim/Projects/testLog/testLog2/App_Packages/LibLog.4.2/LibLog.cs:612 
  at testLog2.Logging.LogProvider.ResolveLogProvider () [0x00033] in /Users/tim/Projects/testLog/testLog2/App_Packages/LibLog.4.2/LibLog.cs:637 
```
Error is not scene when log4net is not in the project. 

----

The issue might be some #if !NETSTANDARD1_3 issues in https://github.com/apache/log4net/blob/trunk/src/Config/BasicConfigurator.cs or elsewhere in log4net. Seems that the repository now has to be set, and maybe LibLog cannot retrieve?

**Comment by damianh** [comment]
Will tackle this in v5 and line up with NetStandard 2.

**Comment by damianh** [comment]
It appears Log4Net on netstandard requires creation of a repository. I believe I have fixed it in this PR https://github.com/damianh/LibLog/pull/149/files#diff-6982fa65e3f004042baeb61ea70f44fdR1335 where I create a repository from the log4net libaray. (`Heirarchy` I think)

**Comment by damianh** [comment]
master branch has this fixed with the move to netstandard #149. Please comment if it remains an issue after v5.0.0 is published.
